src/subcommands/update.go

```diff
@@ -0,0 +1,124 @@
+package subcommands
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+
+	pb "accretional.com/semantifly/proto/accretional.com/semantifly/proto"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type UpdateArgs struct {
+	Name       string
+	IndexPath  string
+	DataType   string
+	SourceType string
+	UpdateCopy string
+	DataURI    string
+}
+
+func Update(u UpdateArgs) {
+	indexFilePath := path.Join(u.IndexPath, indexFile)
+	var index pb.Index
+
+	if err := readIndex(indexFilePath, &index); err != nil {
+		fmt.Printf("Failed to read the index file: %v", err)
+		return
+	}
+
+	if err := updateIndex(&index, &u); err != nil {
+		fmt.Printf("Failed to update the index entry %s: %v", u.Name, err)
+	}
+
+	if err := writeIndex(indexFilePath, &index); err != nil {
+		fmt.Printf("Failed to write to the index file: %v", err)
+		return
+	}
+
+	if u.UpdateCopy == "true" {
+		if err := updateCopy(&u); err != nil {
+			fmt.Printf("Failed to update the copy of the source file: %v", err)
+			return
+		}
+	}
+
+	fmt.Printf("Index %s updated successfully to URI %s\n", u.Name, u.DataURI)
+}
+
+func updateIndex(index *pb.Index, u *UpdateArgs) error {
+	for i, entry := range index.Entries {
+		if entry.Name == u.Name {
+			index.Entries[i].URI = u.DataURI
+
+			if u.DataType != "" {
+				if dataType, err := parseDataType(u.DataType); err != nil {
+					return fmt.Errorf("error in parsing DataType: %v", err)
+				} else {
+					index.Entries[i].DataType = dataType
+				}
+			}
+
+			if u.SourceType != "" {
+				if sourceType, err := parseSourceType(u.SourceType); err != nil {
+					return fmt.Errorf("error in parsing SourceType: %v", err)
+				} else {
+					index.Entries[i].SourceType = sourceType
+				}
+			}
+
+			index.Entries[i].LastRefreshedTime = timestamppb.Now()
+
+			u.SourceType = pb.SourceType_name[int32(index.Entries[i].SourceType)]
+			u.DataType = pb.DataType_name[int32(index.Entries[i].DataType)]
+
+			return nil
+		}
+	}
+
+	return fmt.Errorf("entry %s not found", u.Name)
+}
+
+func updateCopy(u *UpdateArgs) error {
```

Comment:
When you refactor/clean things up after this PR, I think we should consolidate this with the add implementation

Reply:
Yeah I will add this to utils and let both add and update call this.